### PR TITLE
DAOS-8526 test: Make changes to allow avocado to run in env besides CI 

### PR DESCRIPTION
**Pull request by mjean308**
remove the avocado dependency on openmpi in TestWithoutServers
add env to job manager mpirun cmdline using either -x (orterun) or -genv (hydra)
Add ability to use any valid mpi module name in load_module 
Remove mpiio_utils.mpich_installed; job manger handles installing mpich
Signed-off-by: Maureen Jean <maureen.jean@intel.com>

### CODE REVIEW

**Review by shimizukko — DISMISSED** [review]

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Review by shimizukko — DISMISSED** [review]

File: src/tests/ftest/util/job_manager_utils.py [hunk 1]
```diff
@@ -247,6 +300,7 @@ def __init__(self, job, subprocess=False):
         self.ompi_server = FormattedParameter("--ompi-server {}", None)
         self.working_dir = FormattedParameter("-wdir {}", None)
         self.tmpdir_base = FormattedParameter("--mca orte_tmpdir_base {}", None)
+        self.path = path
```
Comment (phender) [inline src/tests/ftest/util/job_manager_utils.py:303]: This is already defined as `self._path` in `CommandWithParameters.__init__()`.  Please remove.
```suggestion
```

File: src/tests/ftest/util/job_manager_utils.py [hunk 2]
```diff
@@ -304,6 +358,15 @@ def assign_environment_default(self, env_vars):
...
         self.export.update_default(env_vars.get_list())
 
+    def get_lib_path(self):
+        """Get path to mpi lib.
+
+        Returns: path to mpi lib
+        """
+        if not self.path:
+            raise CommandFailure("MPI path is not defined")
+        return os.path.join(os.path.split(self.path)[0], 'lib')
```
Comment (phender) [inline src/tests/ftest/util/job_manager_utils.py:368]:
```suggestion
        return os.path.join(os.path.split(self._path)[0], 'lib')
```

File: src/tests/ftest/util/job_manager_utils.py [hunk 3]
```diff
@@ -304,6 +358,15 @@ def assign_environment_default(self, env_vars):
         """
         self.export.update_default(env_vars.get_list())
 
+    def get_lib_path(self):
+        """Get path to mpi lib.
+
+        Returns: path to mpi lib
+        """
+        if not self.path:
```
Comment (phender) [inline src/tests/ftest/util/job_manager_utils.py:366]:
```suggestion
        if not self._path:
```

File: src/tests/ftest/util/job_manager_utils.py [hunk 4]
```diff
@@ -304,6 +358,15 @@ def assign_environment_default(self, env_vars):
...
+            raise CommandFailure("MPI path is not defined")
```
Comment (phender) [inline src/tests/ftest/util/job_manager_utils.py:367]: Given the different paths we have in play here, might I suggest a slightly more clear error message:
```suggestion
            raise CommandFailure("Unable to obtain MPI library path: Full path for {} undefined!".format(self.command))
```

File: src/tests/ftest/util/job_manager_utils.py [hunk 5]
```diff
@@ -304,6 +358,15 @@ def assign_environment_default(self, env_vars):
         """
         self.export.update_default(env_vars.get_list())
 
+    def get_lib_path(self):
+        """Get path to mpi lib.
+
+        Returns: path to mpi lib
```
Comment (phender) [inline src/tests/ftest/util/job_manager_utils.py:364]:
```suggestion
        Returns:
            str: mpi library path

```

File: src/tests/ftest/util/job_manager_utils.py [hunk 6]
```diff
@@ -304,6 +358,15 @@ def assign_environment_default(self, env_vars):
         """
         self.export.update_default(env_vars.get_list())
 
+    def get_lib_path(self):
```
Comment (phender) [inline src/tests/ftest/util/job_manager_utils.py:361]: Should this be moved to the common `JobManager` class? Or maybe a `class MpiJobManager(JobManager):` class that then `Mpirun` and `Orterun` inherit?

File: src/tests/ftest/util/job_manager_utils.py [hunk 7]
```diff
@@ -402,7 +469,16 @@ def assign_environment_default(self, env_vars):
             env_vars (EnvironmentVariables): the environment variables to
                 assign as the default
         """
-        self.envlist.update_default(env_vars.get_list())
+        self.genv.update_default(env_vars.get_list())
+
+    def get_lib_path(self):
```
Comment (phender) [inline src/tests/ftest/util/job_manager_utils.py:474]: Same edits from `Orterun.get_lib_path()` would apply here unless its made common.

**Review by phender — CHANGES_REQUESTED** [review]
In addition the following tests should be run to verify code changes:
- src/tests/ftest/daos_perf/large.py
- src/tests/ftest/daos_perf/small.py
- src/tests/ftest/daos_racer/parallel.py
- src/tests/ftest/daos_vol/bigio.py
- src/tests/ftest/nvme/enospace.py
- src/tests/ftest/nvme/fragmentation.py
- src/tests/ftest/nvme/pool_capacity.py
- src/tests/ftest/server/metadata.py

**Review by shimizukko — APPROVED** [review]

**Review by phender — DISMISSED** [review]

**Review by shimizukko — DISMISSED** [review]

**Review by shimizukko — APPROVED** [review]

**Review by phender — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

**Comment by mjean308** [comment]
5 failures in PR
Centos8: svc/PoolSVC - pool query timeout after rebuild - DAOS-8495
HW small: bigio - hang - known issue  DAOS-6654
HW medium: osa/offline drain snapshot:  network issue
general_utils.DaosTestError: Timeout detected running 'clush -S -v -w wolf-122,wolf-123,wolf-124 /usr/bin/mkdir -p /var/tmp/daos_testing/test_osa_offline_drain_after_snapsot' with a 10s timeout
HW large: soak smoke:    Failed in TearDown: network issue
Timeout detected running 'scp wolf-142:/var/tmp/daos_testing/daos*.log.*   timeout 120secs
HW large: server/metadata.py - DAOS-9078

**Pull request merged**